File: 2022/day02/part1.rb
# frozen_string_literal: true

rounds = File
         .readlines('input.txt', chomp: true)
         .map { |l| l.split(' ') }

win =  %w[AY BZ CX] # 6
draw = %w[AX BY CZ] # 3
lose = %w[AZ BX CY] # 0

result = rounds.sum do |round|
  x1 =
    case round[1]
    when 'X'
      1
    when 'Y'
      2
    when 'Z'
      3
    end

  x2 =
    if win.include?(round.join)
      6
    elsif draw.include?(round.join)
      3
    elsif lose.include?(round.join)
      0
    end

  x1 + x2
end

puts result
# 11841

File: 2022/day02/part1.rs
fn main() {
    let rounds: Vec<Vec<&str>> = include_str!("input.txt")
        .trim_end()
        .lines()
        .map(|l| l.split(' ').collect())
        .collect();

    let win = ["AY", "BZ", "CX"]; // 6
    let draw = ["AX", "BY", "CZ"]; // 3
    let lose = ["AZ", "BX", "CY"]; // 0

    let result: u32 = rounds
        .iter()
        .map(|round| {
            let x1 = match round[1] {
                "X" => 1,
                "Y" => 2,
                "Z" => 3,
                &_ => 0,
            };

            let combination = round.join("");

            // https://doc.rust-lang.org/std/vec/struct.Vec.html#method.contains

            let x2 = if win.iter().any(|e| e == &combination) {
                6
            } else if draw.iter().any(|e| e == &combination) {
                3
            } else if lose.iter().any(|e| e == &combination) {
                0
            } else {
                0
            };

            x1 + x2
        })
        .sum();

    println!("{:?}", result);
}

// 11841
